data/transactions/logic/jsonspec.md

```diff
@@ -0,0 +1,109 @@
+## JSON Spec
+
+A valid JSON text must follow the grammar defined in <a href="https://www.rfc-editor.org/rfc/rfc7159.html">RFC7159</a>
+
+Additional specifications used by **json_ref** that are extensions to the RFC7159 grammar are listed below.
+
+### File Encoding
+- Only utf-8 encoded are accepted
+- The byte order mark (BOM), "\uFEFF", is not allowed at the beginning of a JSON text
+- Raw non-unicode characters not accepted
+#### Invalid JSON text
+```json
+\uFEFF{"key0": 1}
+```
+```json
+{"key0": "\uFF"}
```

Comment:
Very minor, but I think there needs to be 4 characters after every `\u` for it to be valid JSON. E.g. `\u00FF` would work instead

```
> JSON.parse('{"key0": "\uFF"}')
JSON.parse('{"key0": "\uFF"}')
                      ^^^^^^

Uncaught SyntaxError: Invalid Unicode escape sequence
```

Reply:
right. that is an example of invalid json text.

Reply:
Oh, my mistake